src/common/utils/http.ts

```diff
@@ -66,21 +67,36 @@ export async function httpGet<TBody extends JsonMap | string = string , TData =
             encoding: undefined,
             headers,
         },
-    ) as TRequestCoreOptionsRequiredUriUrl;
+    );
 
-    const promisifiedRequest = promisify<TRequestCoreOptionsRequiredUriUrl, request.Response>(request);
-    const response = await promisifiedRequest(requestOptions);
-
-    const result = {
-        isFailure: response.statusCode < 200 || response.statusCode >= 300,
-        isSuccess: response.statusCode >= 200 && response.statusCode < 300,
-        url,
-        responseUrl: response.url,
-        statusCode: response.statusCode,
-        body: response.body,
-        data: callback ? undefined : response.body,
-        contentType: response.caseless.get("Content-Type"),
-    };
+    const result = await (async (opt): Promise<IHttpGetResult<TBody, TData>> =>
```

Comment:
Just out of interest: why remove `promisify`? Why not just `try {} catch(err) {}` the `err` (Promise rejection) which is exactly the same as the `err` parameter of the `request` callback? (then, from inside the `catch(err)` body, treat `ETIMEDOUT` in a specific manner, and `throw` any other `err`)

Note that I am "playing devil's advocate" here, because personally I actually prefer to **avoid** using `util.promisify` which; in my opinion; doesn't add value to the perfectly-fine callback-based mechanism of the original `request` API (in this particular case, not generally-speaking).

So I am not asking because I want `util.promisify` to be re-instated, but because I want to better understand the technical rationale for its removal.

Thank you.